### PR TITLE
added `cupy.format_float_positional`

### PR DESCRIPTION
**Pull request by amanchhaparia**
Hello,
I recently came across the **CUPY** and found it interesting. I am excited and want to contribute to the cupy. 

I have implement the `cupy.format_float_positional` as requested in #6078,
looking for a review and further guidance.

Thank you.

### CODE REVIEW

File: cupy/_io/formatting.py [hunk 1]
```diff
@@ -52,3 +52,12 @@ def array2string(a, *args, **kwargs):
 
     """
     return numpy.array2string(cupy.asnumpy(a), *args, **kwargs)
+
+
+def format_float_positional(a, *args, **kwargs):
+    """Returns the decimal string in positional notation
+
```
Comment (kmaehashi) [inline cupy/_io/formatting.py:59]: * Keep the same signature and docstring as NumPy.
* Add notes about arguments.

```suggestion
def format_float_positional(x, *args, **kwargs):
    """Format a floating-point scalar as a decimal string in positional
    notation.
    
    See :func:`numpy.format_float_positional` for the list of arguments.

```

File: cupy/_io/formatting.py [hunk 2]
```diff
@@ -52,3 +52,12 @@ def array2string(a, *args, **kwargs):
...
+    .. seealso:: :func:`numpy.format_float_positional`
+
+    """
+    return numpy.format_float_positional(cupy.asnumpy(a), *args, **kwargs)
```
Comment (kmaehashi) [inline cupy/_io/formatting.py:63]:
```suggestion
    return numpy.format_float_positional(cupy.asnumpy(x), *args, **kwargs)
```

File: docs/source/reference/io.rst [hunk 3]
```diff
@@ -34,6 +34,7 @@ String formatting
 .. autosummary::
    :toctree: generated/
 
+   format_float_positional
```
Comment (kmaehashi) [inline docs/source/reference/io.rst:37]: Please move this to the line after `array_str` to be consistent with NumPy's reference.
https://numpy.org/doc/stable/reference/routines.io.html

File: tests/cupy_tests/io_tests/test_formatting.py [hunk 4]
```diff
@@ -23,3 +23,9 @@ def test_array2string(self):
         a = testing.shaped_arange((2, 3, 4), cupy)
         b = testing.shaped_arange((2, 3, 4), numpy)
         return cupy.array2string(a) == numpy.array2string(b)
+
```
Comment (kmaehashi) [inline tests/cupy_tests/io_tests/test_formatting.py:26]: Please add a test with python scalar (e.g. `cupy.format_float_positional(1.0)`).

**Review by kmaehashi — CHANGES_REQUESTED** [review]
Thanks!

File: tests/cupy_tests/io_tests/test_formatting.py (hunk 4 above)
Comment (amanchhaparia) [reply tests/cupy_tests/io_tests/test_formatting.py:26]: @kmaehashi, added a test with python scalar

File: docs/source/reference/io.rst (hunk 3 above)
Comment (amanchhaparia) [reply docs/source/reference/io.rst]: Noted.

File: cupy/_io/formatting.py (hunk 2 above)
Comment (amanchhaparia) [reply cupy/_io/formatting.py]: Noted.

File: cupy/_io/formatting.py (hunk 1 above)
Comment (amanchhaparia) [reply cupy/_io/formatting.py]: @kmaehashi, I had updated my PR with suggested changes. can you please review them. Thank you

Comment (amanchhaparia) [reply cupy/_io/formatting.py]: Added suggested changes.

**Comment by amanchhaparia** [comment]
@kmaehashi , I had updated my PR with requested changes can u please review them. Thank You

**Review by kmaehashi — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**